File: Basic/Test_MaximumProductofTwoElementsinanArray.c
// https://leetcode.com/problems/maximum-product-of-two-elements-in-an-array/

#include <stdio.h>

int main()
{
    int size;
    scanf("%d", &size);
    int list[size];
    for (int i = 0; i < size; i++)
    {
        scanf("%d", &list[i]);
    }
    int max;
    for (int i = 0; i < size - 1; i++)
    {
        if (i == 0)
        {
            max = list[i];
        }
        for (int j = i + 1; j < size; j++)
        {
            int product = (list[i] - 1) * (list[j] - 1);
            if (product > max)
            {
                max = product;
            }
        }
    }
    printf("%d", max);
    return 0;
}

File: Basic/Test_MaximumProductofTwoElementsinanArray.py
# https://leetcode.com/problems/maximum-product-of-two-elements-in-an-array/

class Solution:
    def maxProduct(self, nums: list) -> int:
        element = 0
        oneTime = False
        for i in range(len(nums) - 1):
            for j in range(i+1, len(nums)):
                if not oneTime:
                    if i == 0:
                        element = (nums[i] - 1) * (nums[j] - 1)
                        oneTime = True
                element = max(element, (nums[i] - 1) * (nums[j] - 1))
        return element


s = Solution()
print(s.maxProduct(nums=[3, 4, 5, 2]))
# 12
print(s.maxProduct(nums=[1, 5, 4, 5]))
# 16
print(s.maxProduct(nums=[1, 1, 1, 1]))
# 0
print(s.maxProduct(nums=[10, 2, 5, 2]))
# 36
